docs/blueprints/EP025.rst

```diff
@@ -0,0 +1,111 @@
+
+
+Specification
+=============
+
+It is desirable to support ignored loops configuration that can be changed dynamically via API. Since this feature is relatively small and wouldn't require much extra entries or states, then using `topology` switches' and interfaces' metadata storage would simplify the implementation and maintenance since currently adding support to a new storage with storehouse requires a significant effort development-wise. In the future, this can be migrated to another storage (collection) when an alternative is available. The following metadata would be used:
+
+- ``switches.<dpid>.metadata.ignored_loops``:
+
+This is for configuring which loops should be ignored on a ``<dpid>``, the value is a list of list pair of interface port numbers. For instance, if it's desirable to ignore a loop between interface number 5 and 6, and also between port 1 and itself: ``[[5, 6], [1,1]]``
+
+- ``interfaces.<interface_id>.metadata.looped``:
+
+This ``looped`` attribute is for state management, it will contain the following object whenever a loop is in place:
+
+.. code-block:: python3
+
+   {
+     'port_numbers': [<int>, <int>]
+     'detected_at': <str>, in this format %Y-%m-%dT%H:%M:%S
+   }
+
+Looped states should be removed if the interface has been disabled or deactivated, also there should be a periodic routine to remove the ``looped`` metadata interface state if the loop hasn't been detected anymore, this clean up routine should run at a lower pace than of_lldp polling, usually 3-4 times slower, and if the loop hasn't been seen within this time frame again then it would assume that the loop has stopped.
+
+The following options would be configurable on settings:
+
+- ``LLDP_LOOP_ACTIONS``:
+
+This is for configuring which actions should be taken once a loop is found. The actions are a list of strings, ``log`` and ``disable`` are both supported. The ``log`` action will log an warning message informing that a loop was detected on a dpid and on which interfaces. The ``disable`` action will disable an interface via the ``topology`` NApp, and eventually enable it again if the loop stops. For instance, ``['log', 'disable']`` would apply both actions to any loop that isn't ignored and is found. The default actions would be ``['log']`` based on what's expected to be used in production in this first iteration.
+
+- ``LOOP_LOG_EVERY``:
+
+This is for configuring which frequency in seconds the `log` action should emit a log whenever a loop is found, by default it's 60 seconds, this is for avoiding a flood of messages.
```

Comment:
Yes, @jab1982, it's customizable in the settings, with 60 seconds, every 60 seconds it would log a message similar to this one: 

```
kytos $> 2022-03-02 11:37:41,389 - WARNING [kytos.napps.kytos/of_lldp] (ThreadPoolExecutor-0_5) LLDP loop detected on switch: 00:00:00:00:00:00:00:01, interfaces: ['s1-eth1', 's1-eth2'], port_numbers: [1, 2]
```

Let me know which frequency it should be shipped in the settings by default if you have any other value in mind.

Reply:
I would say every 300 seconds or more.

Reply:
Right. So, let's use 300 by default. Done here https://github.com/kytos-ng/kytos/pull/181/commits/4f29a85834b54b18defe00007c1f326d1259dcc7.